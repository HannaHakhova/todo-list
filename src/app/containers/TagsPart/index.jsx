import React, {Component} from 'react';
import {Segment, Header} from 'semantic-ui-react';
import TagList from 'components/TagList';
import TagInput from 'components/TagInput';
import {arrayOf, shape, number, string, func} from 'prop-types';

class TagsPart extends Component {
    constructor(props) {
        super(props);
        this.state = {
            tags: []
        };
    }

    componentWillReceiveProps = nextProps => {
        this.prepareTags(nextProps.tags, nextProps.todosTags);
    }

    prepareTags = (tags, todosTags) => {
        const newTags = tags.map(x => ({...x, count: todosTags.filter(y => y.tagId === x.id).length}));
        this.setState({tags: newTags});
    }

    render() {
        return (
            <div>
                <br />
                <Segment size='mini' textAlign='left' padded='very' >
                    <Header textAlign='center' as='h2'>Tags </Header>
                    <TagInput handleSubmit={this.props.createTag} />
                    <TagList tags={this.state.tags} handleDelete={this.props.deleteTag} />
                </Segment>
            </div>
        );
    }
}
TagsPart.propTypes = {
    tags: arrayOf(shape({
        id: string.isRequired,
        title: string.isRequired}).isRequired).isRequired,
    todosTags: arrayOf(shape({
        id: number.isRequired,
        taskId: string.isRequired,
        tagId: string.isRequired}).isRequired).isRequired,
    createTag: func.isRequired,
    deleteTag: func.isRequired
};

export default TagsPart;

