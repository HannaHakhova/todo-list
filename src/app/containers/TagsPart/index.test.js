/* eslint-disable react/jsx-filename-extension */

import React from 'react';
import {shallow} from 'enzyme';
import TagsPart from './index';

jest.mock('api', () => ({
    getAllData: () => new Promise(resolve => {
        const response = {
            data: [
                {
                    id: '94f3ab7b-f256-4d40-8f98-d6a7c625d816',
                    title: 'Berkshire'
                },
                {
                    id: 'b6fadd6b-2977-4797-8773-ba61da28311b',
                    title: 'Administrator'
                }
            ]
        };
        resolve(response);
    }),
    createData: () => {},
    deleteData: () => {},
    updateData: () => {}
}));

describe('TagsPart', () => {
    it('matches expected snapshot', () => {
        const wrapper = shallow(<TagsPart />);
        expect(wrapper).toMatchSnapshot();
    });

    it('matches expected state ', done => {
        const wrapper = shallow(<TagsPart />);
        const tags = [
            {
                id: '94f3ab7b-f256-4d40-8f98-d6a7c625d816',
                title: 'Berkshire'
            },
            {
                id: 'b6fadd6b-2977-4797-8773-ba61da28311b',
                title: 'Administrator'
            }];

        setTimeout(() => {
            expect(wrapper.state().tags).toEqual(tags);
            done();
        }, 100);
    });
});