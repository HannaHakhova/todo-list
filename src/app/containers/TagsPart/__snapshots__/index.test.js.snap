// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsPart matches expected snapshot 1`] = `
<div>
  <br />
  <Segment
    padded="very"
    size="mini"
    textAlign="left"
  >
    <Header
      as="h2"
      textAlign="center"
    >
      Tags 
    </Header>
    <TagInput />
    <TagList
      tags={Array []}
    />
  </Segment>
</div>
`;
